toolchain/check/handle_class.cpp

```diff
@@ -651,10 +658,37 @@ static auto CheckCompleteClassType(Context& context, Parse::NodeId node_id,
     return CheckCompleteAdapterClassType(context, node_id, class_id, fields_id);
   }
 
+  bool defining_vtable_ptr = class_info.is_dynamic;
+  if (class_info.base_id.is_valid()) {
+    auto base_info = context.insts().GetAs<SemIR::BaseDecl>(class_info.base_id);
+    if (auto* base_class_info = TryGetAsClass(context, base_info.base_type_id);
+        base_class_info && base_class_info->is_dynamic) {
+      defining_vtable_ptr = false;
+    }
+  }
+
+  llvm::SmallVector<SemIR::InstId> object_rep_fields;
+  auto fields = context.inst_blocks().Get(fields_id);
```

Comment:
In the caller of this function, we call `context.args_type_info_stack().Pop();` which will be the point at which the `InstBlock` is allocated. In the case where we define a vtable pointer, we never use that `InstBlock` for anything. I wonder if with some refactoring we could avoid allocating it at all? (Maybe, move the pop from the caller into this function, and add a mechanism to `InstBlockStack` to allow the vptr to be inserted in the right place before we build the block.)

Reply:
Sure - done. (& I think that's the last of the current feedback addressed)